sdk/containerregistry/azure-containerregistry/azure/containerregistry/_download_stream.py

```diff
@@ -0,0 +1,62 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+from typing import Iterable, ContextManager, Callable
+from typing_extensions import Self
+import hashlib
+
+
+class DownloadBlobStream(
+    Iterable[bytes],
+    ContextManager[Self],
+):
+    """Protocol for methods to provide streamed responses."""
+
+    def __init__(self, **kwargs) -> None:
+        self._response: Iterable[bytes] = kwargs.get('response')
+        self._next: Callable[[str], Iterable[bytes]] = kwargs.get('next')
+        self._blob_size: int = kwargs.get('blob_size')
+        self._downloaded: int = kwargs.get('downloaded')
+        self._hasher = hashlib.sha256()
+        self._digest: str = kwargs.get('digest')
+        self._chunk_size: int = kwargs.get('chunk_size')
+
+    def __enter__(self) -> Self:
+        return self
+
+    def __exit__(self, *args) -> None:
+        self.close()
+
+    def __iter__(self) -> Self:
+        return self
+
+    def _yield_data(self) -> bytes:
+        data = next(self._response)
+        self._hasher.update(data)
+        return data
+
+    def _download_chunk(self) -> Iterable[bytes]:
+        end_range = self._downloaded + self._chunk_size
+        range_header = f"bytes={self._downloaded}-{end_range}"
+        next_chunk, headers = self._next(range=range_header)
+        self._downloaded += headers["Content-Length"]
+        return next_chunk
+
+    def __next__(self) -> bytes:
+        try:
+            return self._yield_data()
+        except StopIteration:
+            if self._downloaded >= self._blob_size:
+                computed_digest = "sha256:" + self._hasher.hexdigest()
+                if computed_digest != self._digest:
+                    raise ValueError("The requested digest does not match the digest of the received blob.")
+                raise
+            self._response = self._download_chunk()
+            return self._yield_data()
+
+    def close(self) -> None:
+        # This is not currently supported in Core, and probably needs to be.
+        # self._response.close()
+        pass
```

Comment:
Now that the code autorest regeneration PR has been merged, we need to fix this. Same for async :)